src/lazy.rs

```diff
@@ -2,7 +2,7 @@
 use std::ops::{Deref, DerefMut};
 
 use async_trait::async_trait;
-use tokio::sync::{RwLock, RwLockReadGuard, RwLockWriteGuard};
+use smol::lock::{RwLock, RwLockReadGuard, RwLockWriteGuard};
 
 use crate::Result;
 
```

Comment:
what the `Lazy` structure for?
looks like it can be replaced by `LazyStatic`

Reply:
It is similar with LazyStatic, but it also provide Lock related functionality

Reply:
Why it needs lock? Will the lock impact performance?

Reply:
It is for lease and watch, since lease and watch keep a Tunnel, the lock is to keep the order of Request and Response, it can make sure the keep_alive Request is send before the keep_alive_responses is returned.

Reply:
Add an issue to track this lock, consider refactor it using lock-free data structure.